src/lib/sql-pool.ts

```diff
@@ -0,0 +1,103 @@
+import { createPool, Factory, Pool } from 'generic-pool';
+import { Config, ExasolDriver, websocketFactory } from './sql-client';
+import { ILogger, LogLevel, Logger } from './logger/logger';
+import { QueryResult } from './query-result';
+import { Attributes } from './commands';
+import { CetCancelFunction } from './sql-client.interface';
+import { SQLQueriesResponse, SQLResponse } from './types';
+//https://www.npmjs.com/package/generic-pool
+//https://www.npmjs.com/package/@types/generic-pool
+export interface PoolConfig {
+  min: number;
+  max: number;
+}
+export function getPool(websocketFactory: websocketFactory, config: Partial<Config> & Partial<PoolConfig>, logger: ILogger) {
```

Comment:
```suggestion
function getPool(websocketFactory: websocketFactory, config: Partial<Config> & Partial<PoolConfig>, logger: ILogger) {
```
Does this need to be exported?